src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/PhoneAttribute.cs

```diff
@@ -31,7 +28,100 @@ public override bool IsValid(object value)
             }
 
             var valueAsString = value as string;
-            return valueAsString != null && _regex.Match(valueAsString).Length > 0;
+            if (valueAsString == null)
+            {
+                return false;
+            }
+
+            valueAsString = valueAsString.Replace("+", string.Empty).TrimEnd();
+            valueAsString = RemoveExtension(valueAsString);
+
+            bool digitFound = false;
+            foreach (char c in valueAsString)
+            {
+                if (Char.IsDigit(c))
+                {
+                    digitFound = true;
+                    break;
+                }
+            }
+
+            if (!digitFound)
+            {
+                return false;
+            }
+
+            foreach (char c in valueAsString)
+            {
+                if (!(Char.IsDigit(c)
+                    || Char.IsWhiteSpace(c)
+                    || _additionalPhoneNumberCharacters.IndexOf(c) != -1))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static string RemoveExtension(string potentialPhoneNumber)
+        {
+            var lastIndexOfExtension = potentialPhoneNumber
+                .LastIndexOf(_extensionAbbreviationExtDot, StringComparison.OrdinalIgnoreCase);
+            if (lastIndexOfExtension >= 0)
+            {
+                var extension = potentialPhoneNumber.Substring(
+                    lastIndexOfExtension + _extensionAbbreviationExtDot.Length);
+                if (MatchesExtension(extension))
+                {
+                    return potentialPhoneNumber.Substring(0, lastIndexOfExtension);
+                }
+            }
+
+            lastIndexOfExtension = potentialPhoneNumber
+                .LastIndexOf(_extensionAbbreviationExt, StringComparison.OrdinalIgnoreCase);
+            if (lastIndexOfExtension >= 0)
+            {
+                var extension = potentialPhoneNumber.Substring(
+                    lastIndexOfExtension + _extensionAbbreviationExt.Length);
+                if (MatchesExtension(extension))
+                {
+                    return potentialPhoneNumber.Substring(0, lastIndexOfExtension);
+                }
+            }
+
+            lastIndexOfExtension = potentialPhoneNumber
+                .LastIndexOf(_extensionAbbreviationX, StringComparison.OrdinalIgnoreCase);
+            if (lastIndexOfExtension >= 0)
+            {
+                var extension = potentialPhoneNumber.Substring(
+                    lastIndexOfExtension + _extensionAbbreviationX.Length);
+                if (MatchesExtension(extension))
+                {
+                    return potentialPhoneNumber.Substring(0, lastIndexOfExtension);
+                }
+            }
+
+            return potentialPhoneNumber;
+        }
```

Comment:
Consider refactoring this function something like:
```C#
private static string RemoveExtension(string potentialPhoneNumber)
{
    return
        RemoveExtensionOrNull(potentialPhoneNumber, _extensionAbbreviationExtDot) ??
        RemoveExtensionOrNull(potentialPhoneNumber, _extensionAbbreviationExt) ??
        RemoveExtensionOrNull(potentialPhoneNumber, _extensionAbbreviationX) ??
        potentialPhoneNumber;
}

private static string RemoveExtensionOrNull(string potentialPhoneNumber, string extension)
{
    int lastIndexOfExtension = potentialPhoneNumber.LastIndexOf(extension, StringComparison.OrdinalIgnoreCase);
    if (lastIndexOfExtension >= 0)
    {
        if (MatchesExtension(potentialPhoneNumber.Substring(lastIndexOfExtension + extension.Length)))
        {
            return potentialPhoneNumber.Substring(0, lastIndexOfExtension);
        }
    }
    return null;
}
```
or something like that. (Email-compiled, so take this suggestion with a grain of salt.)